src/rabbit_shovel_worker.erl

```diff
@@ -55,29 +54,62 @@ init([Type, Name, Config0]) ->
                                                                 Config0),
                     Conf
             end,
+    case Name of
+      {VHost, ShovelName} -> rabbit_log:debug("Initialising a Shovel '~s' of type '~s' in virtual host '~s'", [ShovelName, Type, VHost]);
+      ShovelName          -> rabbit_log:debug("Initialising a Shovel '~s' of type '~s'", [ShovelName, Type])
+    end,
+    gen_server2:cast(self(), init),
     {ok, #state{name = Name, type = Type, config = Config}}.
 
 handle_call(_Msg, _From, State) ->
     {noreply, State}.
 
 handle_cast(init, State = #state{config = Config0}) ->
-    Config = rabbit_shovel_behaviour:connect_source(Config0),
-    %% this makes sure that connection pid is updated in case
-    %% any of the subsequent connection/init steps fail. See
-    %% rabbitmq/rabbitmq-shovel#54 for context.
-    gen_server2:cast(self(), connect_dest),
-    {noreply, State#state{config = Config}};
+    try rabbit_shovel_behaviour:connect_source(Config0) of
+      Config ->
+        case maps:get(name, Config) of
+          {VHost, ShovelName} -> rabbit_log:debug("Shovel '~s' in virtual host '~s' connected to source", [ShovelName, VHost]);
+          ShovelName          -> rabbit_log:debug("Shovel '~s' connected to source", [ShovelName])
+        end,
+        %% this makes sure that connection pid is updated in case
+        %% any of the subsequent connection/init steps fail. See
+        %% rabbitmq/rabbitmq-shovel#54 for context.
+        gen_server2:cast(self(), connect_dest),
+        {noreply, State#state{config = Config}}
+    catch _:_ ->
+      case maps:get(name, Config0) of
+        {VHost, ShovelName} -> rabbit_log:error("Shovel '~s' in virtual host '~s' failed to connect to source", [ShovelName, VHost]);
+        ShovelName          -> rabbit_log:error("Shovel '~s' failed to connect to source", [ShovelName])
+      end,
+      {stop, shutdown, State}
+    end;
 handle_cast(connect_dest, State = #state{config = Config0}) ->
-    Config = rabbit_shovel_behaviour:connect_dest(Config0),
-    gen_server2:cast(self(), init_shovel),
-    {noreply, State#state{config = Config}};
+    try rabbit_shovel_behaviour:connect_dest(Config0) of
+      Config ->
+        case maps:get(name, Config) of
+          {VHost, ShovelName} -> rabbit_log:debug("Shovel '~s' in virtual host '~s' connected to destination", [ShovelName, VHost]);
+          ShovelName          -> rabbit_log:debug("Shovel '~s' connected to destination", [ShovelName])
+        end,
+        gen_server2:cast(self(), init_shovel),
+        {noreply, State#state{config = Config}}
+    catch _:_ ->
+      case maps:get(name, Config0) of
```

Comment:
I'd be tempted to refactor these case statements into a reusable function.

Reply:
Done.